modules/gapi/CMakeLists.txt

```diff
@@ -98,14 +99,49 @@ ocv_module_include_directories("${CMAKE_CURRENT_LIST_DIR}/src")
 # (which is opencv_gapi)
 ocv_create_module(ade)
 
+
+
 ocv_add_accuracy_tests()
 # FIXME: test binary is linked with ADE directly since ADE symbols
 # are not exported from libopencv_gapi.so in any form - thus
 # there're two copies of ADE code in memory when tests run (!)
 # src/ is specified to include dirs for INTERNAL tests only.
 if(TARGET opencv_test_gapi)
+
+# This adds custom OpenCL kernel(s) to gapi accuracy tests
+# The similar approach can be used in custom application which uses OpenCV G-API
+  file(GLOB cl_kernels_test
+       "${CMAKE_CURRENT_LIST_DIR}/test/opencl/*.cl"
+  )
+  if(cl_kernels_test)
+    set(OCL_NAME_TEST opencl_kernels_test_${name})
+    add_custom_command(
+      OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${OCL_NAME_TEST}.cpp"  # don't add .hpp file here to optimize build process
+      COMMAND ${CMAKE_COMMAND} "-DMODULE_NAME=${name}" "-DCL_DIR=${CMAKE_CURRENT_LIST_DIR}/test/opencl" "-DOUTPUT=${CMAKE_CURRENT_BINARY_DIR}/${OCL_NAME_TEST}.cpp" -P "${OpenCV_SOURCE_DIR}/cmake/cl2cpp.cmake"
```

Comment:
Not sure if custom application which is using _prebuilt OpenCV_ (as the most users are - e.g. downloaded as a binary package from distro's repository), it's unlikely there's `${OpenCV_SOURCE_DIR}` available in such installation.

I'd recommend to just reading a .cl file from disk - or putting its string version to the source manually (well, then the kernel should be simplier too).

@alalek would there be any issues with such approach? E.g. with respect to CI infrastructure

Reply:
Done

Reply:
Cool